test-suite/compiledboostversion.hpp

```diff
@@ -0,0 +1,31 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2019 Aprexo Limited
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#pragma once 
```

Comment:
In the rest of the library we go with old-style include guards instead.

Reply:
Will do, but is there a plan to do a wholesale migration to `#pragma once`? I would have thought compiler support was pretty universal these days, and it would help a lot with compile time...

Reply:
I wouldn't be opposed if someone does it, but there are probably better uses of people's time...  Does it make a lot of difference in your experience?

Reply:
The official C++ guidelines seem to discourage relying on it, though: see <https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-guards>.  Maybe we could have include guards _and_ pragma once?  But again, it's a lot of work and I'm not sure is worth it.

Reply:
I do recall it having a significant impact on MSVC compile time a large in-house quant library, but I can't recall how much exactly. Looks like [gcc optimises away the old construct anyway](http://gcc.gnu.org/onlinedocs/gcc-2.95.3/cpp_1.html#SEC8), so probably a case of it aint broke, dont fix it